examples/notebooks/armory_colab_example.ipynb

```diff
@@ -0,0 +1,1232 @@
+            "Requirement already satisfied: dill<0.3.8,>=0.3.0 in /home/chris/.virtualenvs/cdao/lib/python3.10/site-packages (from datasets) (0.3.6)\n",
+            "Requirement already satisfied: yarl<2.0,>=1.0 in /home/chris/.virtualenvs/cdao/lib/python3.10/site-packages (from aiohttp->datasets) (1.9.2)\n",
+            "Requirement already satisfied: frozenlist>=1.1.1 in /home/chris/.virtualenvs/cdao/lib/python3.10/site-packages (from aiohttp->datasets) (1.3.3)\n",
+            "Requirement already satisfied: attrs>=17.3.0 in /home/chris/.virtualenvs/cdao/lib/python3.10/site-packages (from aiohttp->datasets) (23.1.0)\n",
+            "Requirement already satisfied: aiosignal>=1.1.2 in /home/chris/.virtualenvs/cdao/lib/python3.10/site-packages (from aiohttp->datasets) (1.3.1)\n",
+            "Requirement already satisfied: multidict<7.0,>=4.5 in /home/chris/.virtualenvs/cdao/lib/python3.10/site-packages (from aiohttp->datasets) (6.0.4)\n",
+            "Requirement already satisfied: charset-normalizer<4.0,>=2.0 in /home/chris/.virtualenvs/cdao/lib/python3.10/site-packages (from aiohttp->datasets) (3.2.0)\n",
+            "Requirement already satisfied: async-timeout<5.0,>=4.0.0a3 in /home/chris/.virtualenvs/cdao/lib/python3.10/site-packages (from aiohttp->datasets) (4.0.2)\n",
+            "Requirement already satisfied: typing-extensions>=3.7.4.3 in /home/chris/.virtualenvs/cdao/lib/python3.10/site-packages (from huggingface-hub<1.0.0,>=0.14.0->datasets) (4.5.0)\n",
+            "Requirement already satisfied: filelock in /home/chris/.virtualenvs/cdao/lib/python3.10/site-packages (from huggingface-hub<1.0.0,>=0.14.0->datasets) (3.12.2)\n",
+            "Requirement already satisfied: certifi>=2017.4.17 in /home/chris/.virtualenvs/cdao/lib/python3.10/site-packages (from requests>=2.19.0->datasets) (2022.12.7)\n",
+            "Requirement already satisfied: urllib3<3,>=1.21.1 in /home/chris/.virtualenvs/cdao/lib/python3.10/site-packages (from requests>=2.19.0->datasets) (1.26.16)\n",
+            "Requirement already satisfied: idna<4,>=2.5 in /home/chris/.virtualenvs/cdao/lib/python3.10/site-packages (from requests>=2.19.0->datasets) (2.10)\n",
+            "Requirement already satisfied: python-dateutil>=2.8.2 in /home/chris/.virtualenvs/cdao/lib/python3.10/site-packages (from pandas->datasets) (2.8.2)\n",
+            "Requirement already satisfied: tzdata>=2022.1 in /home/chris/.virtualenvs/cdao/lib/python3.10/site-packages (from pandas->datasets) (2023.3)\n",
+            "Requirement already satisfied: pytz>=2020.1 in /home/chris/.virtualenvs/cdao/lib/python3.10/site-packages (from pandas->datasets) (2023.3)\n",
+            "Requirement already satisfied: six>=1.5 in /home/chris/.virtualenvs/cdao/lib/python3.10/site-packages (from python-dateutil>=2.8.2->pandas->datasets) (1.16.0)\n"
+          ]
+        }
+      ],
+      "source": [
+        "!pip install armory-library\n",
+        "!pip install datasets"
```

Comment:
We should fix this correctly instead of putting out an example like this. Either armory-library needs to call out `datasets` as a dependency or we should make it an optional dependency that's only required if you're using the `AdversarialDatasetEngine` (I think that's where it gets used).

Reply:
In colab, I am not able to `import datasets` if I don't pip install datasets . That is used here in `dataset = datasets.load_dataset("mnist", split="test")`. So is it best to make datasets a dependency?

Reply:
Ah, good point. I forgot we were using it directly as part of the example (we still have another problem where it's an implicit dependency of armory-library, and I thought that was what was happening here).

Then maybe just an explanation of why it's being installed here. Something like "we install datasets because we will use it later to retrieve a dataset from the HuggingFace hub". It's unfortunate though, because we also need the transformers package to be installed in order to get the model from HF--but that one _is_ a declared dependency of armory-library (for now, anyway).

Reply:
Okay will do thanks that makes sense to me!